src/main/java/com/microsoft/appcenter/event/Event.java

```diff
@@ -12,6 +23,9 @@ private Event(String id, EventType type, String testName, String className, Stri
     }
 
     public static Event createLabel(String id, String testName, String className, String run, String label) {
+        if (null == label || label.length() < 1 || label.length() > MAX_LABEL_SIZE) {
+            throw new IllegalArgumentException("labels must be non-empty strings of length <= " + MAX_LABEL_SIZE);
+        }
         Event event = new Event(id, EventType.label, testName, className, run);
```

Comment:
As noted on the appium pr I would rather be truncating at 160 chars, but if you prefer a hard limit of 256 I am fine with that also. There is a technical reason why << 256 is preferable. On some filesystems paths are limited to 256 chars, and we see labels and other identifiers ending up in paths from time to time... But no biggie, just a note.